### PR TITLE
Updated level to v3.0.0

### PR DESCRIPTION
**Pull request by luandro**
`Leveldown` v3.0.0 can compile for Android.

### CODE REVIEW

**Comment by arj03** [comment]
Not that we shouldn't merge this, but I have been testing 3.0 and performance seems to be worse. Havn't digged enough into why it is exactly, but just a heads up.

**Comment by dominictarr** [comment]
@arj03 how much worse? I think running on android outweighs that, anyway.

**Comment by arj03** [comment]
Can't remember on the top of my head but I think it was something like 10-20%. I actually tested the upgrade because I thought it would be a win 😄 But yeah as said, please don't hold back on the merge. I'll spend some time soonish digging into why it is slower.

**Comment by dominictarr** [comment]
merged into 3.0.2

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   "dependencies": {
     "charwise": "^3.0.1",
     "explain-error": "^1.0.4",
-    "level": "^1.7.0",
+    "level": "git://github.com/luandro/level",
```
[the remark this answers is not in the document]
Comment (luandro) [reply package.json]: There's a problem with level's `postinstall` script when running on mobile. I'm not sure what to do about that.

[the remark this answers is not in the document]
Comment (cryptix) [reply package.json]: Okay. I just didn't assume this from the commit messages and pr thread.

very much love for jumping though these hoops!

[the remark this answers is not in the document]
Comment (luandro) [reply package.json]: Yea sorry, should have been more explicit about the change.

**Comment by arj03** [comment]
I ran the bench.js script. I saw quite a bit of variance so bumped the number of runs in test-flumeview-index to 50000. After this I don't see much difference so that's really good! I think what I saw before was related to startup and then the variance. I've added a patch here: https://github.com/flumedb/test-flumeview-index/pull/1. 

And secondly its really nice that my machine don't have to recompile leveldb as there are no prebuilt versions :smile: :+1: @luandro

**Comment by dominictarr** [comment]
@luandro did you post an issue on level about the postinstall script? what is the problem with it? maybe there is a simple way to change it so that it works - this will effect other people too and I'm sure level will be happy to announce it works on moblie.

**Comment by luandro** [comment]
@dominictarr I haven't. The postinstall script throws an error on mobile, I'll reproduce it again in order to post the issue there. Thanks for the heads up.

**Comment by luandro** [comment]
Here's the error I get when trying to compile Level that's using the Open Collective `postinstall` script:

```
> level@3.0.0 postinstall /Users/projects/ssb-nodejs-mobile/android/build/nodejs-native-assets-temp-build/nodejs-native-assets-armeabi-v7a/nodejs-project/node_modules/level
> opencollective postinstall

module.js:549
    throw err;
    ^

Error: Cannot find module '/Users/projects/ssb-nodejs-mobile/android/build/nodejs-native-assets-temp-build/nodejs-native-assets-armeabi-v7a/nodejs-project/node_modules/commands/postinstall.js'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/projects/ssb-nodejs-mobile/android/build/nodejs-native-assets-temp-build/nodejs-native-assets-armeabi-v7a/nodejs-project/node_modules/.bin/opencollective:57:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
npm verb lifecycle level@3.0.0~postinstall: unsafe-perm in lifecycle true
npm verb lifecycle level@3.0.0~postinstall: PATH: /Users/.nvm/versions/node/v8.10.0/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/projects/ssb-nodejs-mobile/android/build/nodejs-native-assets-temp-build/nodejs-native-assets-armeabi-v7a/nodejs-project/node_modules/level/node_modules/.bin:/Users/projects/ssb-nodejs-mobile/android/build/nodejs-native-assets-temp-build/nodejs-native-assets-armeabi-v7a/nodejs-project/node_modules/.bin:/usr/local/bin:/usr/local/sbin:/Users/.nvm/versions/node/v8.10.0/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/git/bin:/Users/Library/Android/sdk/tools:/Users/Library/Android/sdk/platform-tools:/usr/local/go/bin
npm verb lifecycle level@3.0.0~postinstall: CWD: /Users/projects/ssb-nodejs-mobile/android/build/nodejs-native-assets-temp-build/nodejs-native-assets-armeabi-v7a/nodejs-project/node_modules/level
npm info lifecycle level@3.0.0~postinstall: Failed to exec postinstall script
npm verb stack Error: level@3.0.0 postinstall: `opencollective postinstall`
npm verb stack Exit status 1
npm verb stack     at EventEmitter.<anonymous> (/Users/.nvm/versions/node/v8.10.0/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:283:16)
npm verb stack     at emitTwo (events.js:126:13)
npm verb stack     at EventEmitter.emit (events.js:214:7)
npm verb stack     at ChildProcess.<anonymous> (/Users/.nvm/versions/node/v8.10.0/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
npm verb stack     at emitTwo (events.js:126:13)
npm verb stack     at ChildProcess.emit (events.js:214:7)
npm verb stack     at maybeClose (internal/child_process.js:925:16)
npm verb stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
```

**Comment by dominictarr** [comment]
@Luandro that error is fixed in level@3.0.1

**Pull request merged**